### ISSUE TITLE
Don't run tests for non-code changes

### ISSUE THREAD

**Issue by wd60622** [pymc-labs/pymc-marketing#884]
Does it make sense to only run the test suite if the code has changed? If not running the tests if the case of only docstring changes is possible, is this desirable?

Something like this might be able to be done:
```yaml
on:
  push:
    paths:
      - 'pymc_marketing/**/*.py'
  pull_request:
    paths:
      - 'pymc_marketing/**/*.py'
```

**Comment by dandeandean** [comment]
Hey @wd60622 as it happens I've been looking gh actions recently. We can see something similar in Github docs [here](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-your-pull_request-workflow-based-on-files-changed-in-a-pull-request), which run on any matching paths.

 In this case it'd be: 
```yml
on:
  pull_request:
    branches: [main]
    paths: '**.py'
  push:
    branches: [main]
    paths: '**.py'
```
Furthermore, you can add a `paths-ignore`, as described [here](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-excluding-paths). This may not be the way to go because it won't run if any of the paths match the pattern given.

**Comment by wd60622** [comment]
> Hey @wd60622 as it happens I've been looking gh actions recently. We can see something similar in Github docs [here](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-your-pull_request-workflow-based-on-files-changed-in-a-pull-request), which run on any matching paths.
> 
> 
> 
>  In this case it'd be: 
> 
> ```yml
> 
> on:
> 
>   pull_request:
> 
>     branches: [main]
> 
>     paths: '**.py'
> 
>   push:
> 
>     branches: [main]
> 
>     paths: '**.py'
> 
> ```
> 
> Furthermore, you can add a `paths-ignore`, as described [here](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-excluding-paths). This may not be the way to go because it won't run if any of the paths match the pattern given. 

Very nice @dandeandean 
Do these file conditions happen at the file level or the individual level? I believe some of the same file's jobs target non-source code (notebooks, general formatting)

**Comment by wd60622** [comment]
What are your thoughts @juanitorduz 
Seems like there are some prs that affect specfic elements of the repo. For instance, documentation / notebooks. Would saving 18+ mins of test suite help with dev experience?

**Comment by wd60622** [comment]
The ci.yml file has a few purposes: 
- Run pre-commit
- Run tests (slow and fast)  & check they both sucessfully run
- Check for notebooks running

Running the pre-commit is fast and not much of a concern for me. Reasonable with notebooks too
The tests are the main concern. 

Separating them out into test.yml and putting this conditional filter is where my head is at. Would like to hear other thoughts on this and if there are pros and cons to this approach.

**Comment by wd60622** [comment]
I've merged in #886 which uses the `paths: ["pymc_marketing/**"]` for generating the UML
